File: logger.py
"""Logger sederhana buat userbot (pakai logging standar)."""
import logging
import os
import sys

_LEVEL = (os.getenv("LOG_LEVEL", "INFO") or "INFO").upper()


class _Fmt(logging.Formatter):
    _COLORS = {
        "DEBUG": "\033[90m",
        "INFO": "\033[36m",
        "WARNING": "\033[33m",
        "ERROR": "\033[31m",
        "CRITICAL": "\033[41m",
    }
    _RESET = "\033[0m"

    def format(self, record: logging.LogRecord) -> str:
        color = self._COLORS.get(record.levelname, "")
        record.levelname = f"{color}{record.levelname}{self._RESET}"
        return super().format(record)


def get_logger(name: str) -> logging.Logger:
    log = logging.getLogger(name)
    if not log.handlers:
        h = logging.StreamHandler(sys.stdout)
        h.setFormatter(_Fmt("%(asctime)s %(levelname)s %(name)s: %(message)s", "%H:%M:%S"))
        log.addHandler(h)
        log.setLevel(_LEVEL)
        log.propagate = False
    return log


logger = get_logger("telegrambot-ai")

File: src/observability/logger.ts
import pino from "pino";
import { getEnv } from "../config/env.js";

let logger: pino.Logger | null = null;

export function getLogger(): pino.Logger {
  if (logger) return logger;
  let level = "info";
  try {
    level = getEnv().LOG_LEVEL;
  } catch {
    level = process.env.LOG_LEVEL ?? "info";
  }
  logger = pino({
    level,
    formatters: {
      level: (label) => ({ level: label }),
    },
  });
  return logger;
}
